client/my-sites/activity/filterbar/action-type-selector.jsx

```diff
@@ -235,40 +233,54 @@ export class ActionTypeSelector extends Component {
 	}
 }
 
-const mapStateToProps = ( state, { siteId, filter } ) => {
-	const activityTypes = siteId && requestActivityActionTypeCounts( siteId, filter );
-	const selectedState = filter && filter.group;
-	return {
-		activityTypes: ( siteId && activityTypes.data ) || [],
-		selectedState,
-	};
-};
-
-const mapDispatchToProps = ( dispatch ) => ( {
-	selectActionType: ( siteId, group, allTypes ) => {
-		if ( 0 === group.length ) {
-			return dispatch(
-				withAnalytics(
-					recordTracksEvent( 'calypso_activitylog_filterbar_reset_type' ),
-					updateFilter( siteId, { group: null, page: 1 } )
+const activityCountsQueryKey = ( siteId, filter ) => [
+	'activity-log-counts',
+	siteId,
+	filter.before ?? '',
+	filter.after ?? '',
+	filter.on ?? '',
+];
+const withActivityTypes = ( WrappedComponent ) => ( props ) => {
+	const { siteId, filter } = props;
+	const { data } = useQuery(
+		activityCountsQueryKey( siteId, filter ),
+		() =>
+			wpcom.req
+				.get(
+					{ path: `/sites/${ siteId }/activity/count/group`, apiNamespace: 'wpcom/v2' },
+					filterStateToApiQuery( filter, false )
 				)
-			);
-		}
-		const eventProps = { num_groups_selected: group.length };
-		allTypes.forEach(
-			( type ) => ( eventProps[ 'group_' + type.key ] = group.includes( type.key ) )
-		);
-		eventProps.num_total_activities_selected = allTypes.reduce( ( accumulator, type ) => {
-			return group.includes( type.key ) ? accumulator + type.count : accumulator;
-		}, 0 );
+				.then( fromActivityTypeApi ),
+		{ enabled: !! siteId }
```

Comment:
In the previous version this seemed to update every 10 seconds, should we keep that?

Reply:
Good point, I replicated that behavior by adding a `staleTime` query option. Although I'm not sure it's really needed.

Reply:
Guess the idea is to reproduce what we had before. But yeah, not sure it's really needed to re-query so intensively 🤷